suricata/update/main.py

```diff
@@ -984,14 +990,19 @@ def load_sources(suricata_version):
             params.update(internal_params)
             if "url" in source:
                 # No need to go off to the index.
-                url = (source["url"] % params, source.get("http-header"))
+                http_header = source.get("http_header")
+                no_checksum = source.get("no-checksum")
+                url = (source["url"] % params, http_header, no_checksum)
                 logger.debug("Resolved source %s to URL %s.", name, url[0])
             else:
                 if not index:
                     raise exceptions.ApplicationError(
                         "Source index is required for source %s; "
                         "run suricata-update update-sources" % (source["source"]))
-                url = index.resolve_url(name, params)
+                http_header = source.get("http_header")
+                no_checksum = source.get("no-checksum")
```

Comment:
This doesn't work when the no-checksum option is specified in the configuration file, as you are getting the no-checksum value from the source configuration, which is fine for an `add-source` type of source, but not when the configuration details are coming from the index.  With the current state of the code you'll probably have to do something more like:
```
                source_config = index.get_source_by_name(name)
                try:
                    no_checksum = source_config["no-checksum"]
                except:
                    no_checksum = False
```

Reply:
@jasonish  Thanks for the feedback. I'll make the requested changes in a new PR.